docs/internals/contributing/writing-code/javascript.txt

```diff
@@ -58,7 +58,81 @@ independently. The Closure Compiler library requires `Java`_ 7 or higher.
 Please don't forget to run ``compress.py`` and include the ``diff`` of the
 minified scripts when submitting patches for Django's JavaScript.
 
+JavaScript tests
+----------------
+
+Django includes JavaScript tests that can easily be run in a browser or, after
+installing dependencies, from the command line.
+
+Writing tests
+-------------
+
+Django's JavaScript tests use `QUnit`_. Here is an example test module::
+
+    module('magicTricks', {
+        beforeEach: function() {
+            var $ = django.jQuery;
+            $('#qunit-fixture').append('<button class="button"></button>');
+        }
+    });
+
+    test('removeOnClick removes button on click', function(assert) {
+        var $ = django.jQuery;
+        removeOnClick('.button');
+        assert.equal($('.button').length === 1);
+        $('.button').click();
+        assert.equal($('.button').length === 0);
+    });
+
+    test('copyOnClick adds button on click', function(assert) {
+        var $ = django.jQuery;
+        copyOnClick('.button');
+        assert.equal($('.button').length === 1);
+        $('.button').click();
+        assert.equal($('.button').length === 2);
+    });
+
+
+Please consult the QUnit documentation for information on the types of
+`assertions supported by QUnit <https://api.qunitjs.com/category/assert/>`_.
+
+Running tests
+-------------
+
+The JavaScript tests may be run from a web browser or from the command-line.
+
+Testing from a web browser
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To run the tests from a web browser, just open up the file
+``js_tests/tests.html`` in your browser.
+
+To measure code coverage when running the tests, you will need to view that
+file over HTTP. To view code coverage:
+
+* Execute ``python -m SimpleHTTPServer`` from the inside the ``js_tests``
+  directory
+* Open http://localhost:8000/tests.html in your web browser
```

Comment:
Yes that explanation was wrong.  The documentation now explains that `SimpleHTTPServer` must be started from the top-level directory.  I'm open to suggestions to further fix/clarify the wording.

@oinopion I hadn't realized that `SimpleHTTPServer` changed in Python 3.  Should that be clarified?